File: content/snippets/scala3/interview_lcci/17/16_the_masseuse.scala
object Solution:
  def massage(nums: Array[Int]): Int =
    var f = 0
    var g = 0
    for (x <- nums) {
      val nf = g + x
      val ng = f.max(g)
      f = nf
      g = ng
    }
    f.max(g)

File: content/snippets/cpp/interview_lcci/17/16_the_masseuse.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int massage(vector<int>& nums) {
        int f = 0, g = 0;
        for (int x : nums) {
            int nf = g + x;
            int ng = max(f, g);
            f = nf;
            g = ng;
        }
        return max(f, g);
    }
};
